src/System.Windows.Forms/src/System/Windows/Forms/TextBoxBase.TextBoxBaseUiaTextProvider.cs

```diff
@@ -205,9 +205,12 @@ public override LOGFONTW Logfont
 
         public override string Text
             => Owner is not null && Owner.IsHandleCreated
-                ? Owner.Text
+                ? Owner.PasswordProtect ? _passwordString
+                : Owner.Text
                 : string.Empty;
 
+        private string _passwordString => Owner is not null ? new string(Owner is TextBox ? ((TextBox)Owner).PasswordChar : ((MaskedTextBox)Owner).PasswordChar, Owner.Text.Length) : string.Empty;
```

Comment:
nit: for readability
```suggestion
        private string _passwordString => Owner is not null 
        ? new string(Owner is TextBox textBox ? textBox.PasswordChar : ((MaskedTextBox)Owner).PasswordChar, Owner.Text.Length)
        : string.Empty;
```

Reply:
Done.